eppoclient/value.go

```diff
@@ -14,79 +15,101 @@ const (
 )
 
 type Value struct {
-	valueType    ValueType
-	boolValue    bool
-	numericValue float64
-	stringValue  string
+	ValueType    ValueType `json:"valueType"`
+	BoolValue    bool      `json:"boolValue,omitempty"`
+	NumericValue float64   `json:"numericValue,omitempty"`
+	StringValue  string    `json:"stringValue,omitempty"`
 }
 
 func Null() Value {
-	return Value{valueType: NullType}
+	return Value{ValueType: NullType}
 }
 
 func Bool(value bool) Value {
-	return Value{valueType: BoolType, boolValue: value}
+	return Value{ValueType: BoolType, BoolValue: value}
 }
 
 func Numeric(value float64) Value {
-	return Value{valueType: NumericType, numericValue: value}
+	return Value{ValueType: NumericType, NumericValue: value}
 }
 
 func String(value string) Value {
-	return Value{valueType: StringType, stringValue: value}
+	return Value{ValueType: StringType, StringValue: value}
 }
 
-func (receiver *Value) UnmarshalJSON(data []byte) error {
-	var valueInterface interface{}
-	if err := json.Unmarshal(data, &valueInterface); err != nil {
-		return err
+func (v Value) GetBoolValue() bool {
+	return v.ValueType == BoolType && v.BoolValue
+}
+
+func (v Value) GetStringValue() string {
+	if v.ValueType == StringType {
+		return v.StringValue
 	}
-	*receiver = castInterfaceToValue(valueInterface)
+	return ""
+}
 
-	return nil
+func (v Value) GetNumericValue() float64 {
+	if v.ValueType == NumericType {
+		return v.NumericValue
+	}
+
+	return 0
 }
 
-func castInterfaceToValue(valueInterface interface{}) Value {
```

Comment:
this function was not being re-used so i folded it into the `UnmarshalJSON` method